opensquirrel/writer/writer.py

```diff
@@ -50,20 +50,23 @@ def visit_float(self, f: SupportsFloat) -> str:
         return f"{f.value:.{self.FLOAT_PRECISION}}"
 
     def visit_non_unitary(self, non_unitary: NonUnitary) -> None:
+        non_unitary_name = non_unitary.name
+        qubit_argument = non_unitary.arguments[0].accept(self)  # type: ignore[index]
         if non_unitary.name == "measure":
             bit_argument = non_unitary.arguments[1].accept(self)  # type: ignore[index]
-            qubit_argument = non_unitary.arguments[0].accept(self)  # type: ignore[index]
-            self.output += f"{bit_argument} = {non_unitary.name} {qubit_argument}\n"
+            non_unitary_name = f"{bit_argument} = {non_unitary_name}"
         else:
-            qubit_argument = non_unitary.arguments[0].accept(self)  # type: ignore[index]
-            self.output += f"{non_unitary.name} {qubit_argument}\n"
+            params = [param.accept(self) for param in non_unitary.arguments[1::]]  # type: ignore[index]
+            if params:
+                non_unitary_name = f"{non_unitary_name}({', '.join(params)})"
+        self.output += f"{non_unitary_name} {qubit_argument}\n"
```

Comment:
Suggestion  (not tested):

```
        bit_argument = non_unitary.arguments[1].accept(self)  if non_unitary.name == "measure" else ""  # type: ignore[index]
        bit_section = f"{bit_argument} = " if not bit_argument.empty() elsse ""
        params = [param.accept(self) for param in non_unitary.arguments[1::]]  # type: ignore[index]
        param_section = f"({', '.join(params)})" if params else  ""
        qubit_argument = non_unitary.arguments[0].accept(self)  # type: ignore[index]
        self.output += f"{bit_section}{non_unitary.name}{param_section} {qubit_argument}\n"
```

Reply:
It's nice to generalize the generation of these statements. However, this suggestion does not work because the 1-th argument of a measure statement is the bit value, which is then also interpreted as a parameter.

We will need to revisit this code when we include the support for parameterized non-unitaries (with axis parameters)

Reply:
OK, true, thanks. We could check for `non_unitary_name != "measure"` in the `params =` line, but that is overcomplicating the code.

What I would do then is to set `self.output` in both blocks, mainly to avoid that line 57 (`non_unitary_name` is not really a name there, is a bit argument plus an equals sign plus a name).